### PR TITLE
DEV-665, DEV-686 - run mb, ls, and ptsearch in docker locally

### PR DESCRIPTION
**Pull request by aelkiss**
Needs https://github.com/hathitrust/ls/pull/2

### CODE REVIEW

**Review by respinos — COMMENTED** [review]
General question: do any of the volumes persist docker down/up? Surprised I had to re-stage the item I was using.

`ls` works, but `ptsearch` is missing a grant: `INSERT, UPDATE command denied to user 'mdp-lib'@'172.20.0.8' for table `ht`.`slip_host_control`

`mdp-admin` doesn't have GRANT permissions (for the easy fix)

**Comment by respinos** [comment]
@aelkiss added 

```
GRANT INSERT, UPDATE, DELETE ON `ht`.`slip_host_control` TO 'mdp-lib'@'%';
GRANT INSERT, UPDATE, DELETE ON `ht`.`slip_shard_control` TO 'mdp-lib'@'%';
GRANT INSERT, UPDATE, DELETE ON `ht`.`slip_shard_stats` TO 'mdp-lib'@'%';
GRANT INSERT, UPDATE, DELETE ON `ht`.`slip_indexed` TO 'mdp-lib'@'%';
GRANT INSERT, UPDATE, DELETE ON `ht`.`slip_errors` TO 'mdp-lib'@'%';
GRANT INSERT, UPDATE, DELETE ON `ht`.`slip_rights` TO 'mdp-lib'@'%';
```

and `pt/search` indexing/search works) but `pt/search` needs to be made aware of the port when making search requests (in the reader) and the index/search redrect that `pt/search` does.

**Comment by respinos** [comment]
@aelkiss worked out the `pt/search` redirect issue and can search the book locally.

File: docker/babel-base/Dockerfile [hunk 1]
```diff
@@ -55,7 +57,7 @@ RUN cpanm --notest \
   UUID::Tiny \
   YAML::Any \
   Data::Page
```
[the remark this answers is not in the document]
Comment (aelkiss) [reply docker/babel-base/Dockerfile:59]: Data::Page - yes, I'll fix that; YAML::Any - not that I see. `libyaml-libyaml-perl` installs `YAML::LibYAML` and `YAML::XS`, but not `YAML::Any`.

**Comment by respinos** [comment]
@aelkiss I updated the `nginx` config to send the port to apache which fixes the URLs generated by `mb` and `ls` (verified that `ls` works; `mb` is tricky --- it's generating the correct URLs but I suspect it's having session problems with `localhost`.

This PR can probably be merged.

**Comment by aelkiss** [comment]
OK, I'll give it another try and see if it's working for me, then I'll squash & merge.

**Comment by respinos** [comment]
A few things! To run off `localhost` you'll need a patch to `mdp-lib/Utils.pm` with 

```    
    my $virtual_host = HTTP_hostname();
    return 'localhost' if ( $virtual_host =~ m,^localhost,);
```

to avoid `.localhost`, which doesn't work. You need that for temporary collections to stick for you (they're sticking in the database of course).

Then `mb` needs the same kind of `mbooks_solr_engines` customization that's being done for `ls`.

I'll put up a PR for `mdp-lib` later today.

**Comment by respinos** [comment]
@aelkiss also updated the compose/Dockerfile to use the actual submodule checkouts instead of making the symlink in the dockerfile.

**Comment by aelkiss** [comment]
I tried out the current state and observed the issues with mb that you mentioned, so I wasn't able yet to fully test that the redirects are working as expected in mb. I did see that the radio buttons in ls were working at least, so that's promising.

I also tried using `local.conf` for `mb` and initially that seemed like it might work, so I might revisit that for `ls` as well.

**Comment by aelkiss** [comment]
> General question: do any of the volumes persist docker down/up? Surprised I had to re-stage the item I was using.

Fixed in https://github.com/hathitrust/babel-local-dev/pull/3/commits/93f36fb346171b66e09110216e786b0b1d056727

**Comment by aelkiss** [comment]
Squashed; merging

**Pull request merged**